server/src/main/java/org/elasticsearch/indices/recovery/SnapshotFilesProvider.java

```diff
@@ -52,6 +52,11 @@ public InputStream getInputStreamForSnapshotFile(
                 protected InputStream openSlice(int slice) throws IOException {
                     return container.readBlob(OperationPurpose.SNAPSHOT_DATA, fileInfo.partName(slice));
                 }
+
+                @Override
+                public boolean markSupported() {
+                    return false;
+                }
```

Comment:
I am not sure this is important? Can we remove it? Similar for the next one.

Reply:
There's a question whether we'd like current implementations to have markSupported as true? I originally enabled them in the PR because I believe they are resumable as they read blobs. But @ywangd [mentioned](https://github.com/elastic/elasticsearch/pull/112563#pullrequestreview-2284807106): 

> I wonder whether you have considered adding the changes as a subclass in the stateless codebase? Not that I find any change here is unsafe, but a tighter scope generally feels safer to me.

In light of this, I disabled markSupported in current implementations. They will behave as they were (without resumability). But feel free to tell me if I should go ahead and enable markSupported in all current implementations.

Reply:
I was thinking to add a separate subclass for `SlicedInputStream` and implement mark/reset support in the subclass so that existing usages of `SlicedInputStream` keep unchanged. But I guess it might be challenging to implement the new features in a separate class? The current version works as well. I slightly prefer to have a follow-up PR just to enable the mark/reset support for existing usages since this PR is to support the stateless side changes. Separating them seems to make that concept clearer.

Reply:
Indeed it'd be challenging to have them separate, plus I believe it's useful to have the capability in the core SlicedInputStream since all implementations seem to support mark/reset (even though we do not enable them just to maintain their old behavior just to be safe).